### PR TITLE
Update location card implementations

### PR DESCRIPTION
**Pull request by DukeTax**
* Updates all currently implemented locations.
* Converts implementations to use the cost and/or target API.
* Makes sure the ability can only trigger or initiate if it would change the game state.
* Standardises action labels, prompt titles and chat messages.
* Changes actions to use a handler instead of method, where needed.
* Changes var into let where possible.
* Invokes game.transferGold() instead of two times invoking game.addGold().

### CODE REVIEW

File: server/game/cards/locations/01/castleblack.js [hunk 1]
```diff
@@ -3,37 +3,26 @@ const DrawCard = require('../../../drawcard.js');
...
-            title: 'Kneel this card to stand and give a defending character +2 STR',
-            cost: ability.costs.kneelSelf(),
+            title: 'Stand and give defending character +2 STR',
             condition: () => this.game.currentChallenge,
-            method: 'kneel'
-        });
-    }
+            cost: ability.costs.kneelSelf(),
+            target: {
+                activePromptTitle: 'Select a character',
```
Comment (cryogen) [inline server/game/cards/locations/01/castleblack.js]: Really bugs me that this isn't the default, remind me to change that sometime

**Comment by cryogen** [comment]
There's some conflicts here that need resolving

**Comment by DukeTax** [comment]
Github could use a feature alerting PR owners of when this happens.

**Comment by cryogen** [comment]
Or I could just be more helpful and point it out sooner :)

**Pull request merged**